src/io/usb_serial_jtag.rs

```diff
@@ -0,0 +1,59 @@
+use super::RX_QUEUE;
+use crate::{
+    hal::{
+        prelude::*,
+        usb_serial_jtag::{Instance, UsbSerialJtag},
+    },
+    protocol::InputIO,
+};
+
+impl InputIO for UsbSerialJtag<'_> {
+    fn recv(&mut self) -> u8 {
+        unsafe { while critical_section::with(|_| RX_QUEUE.is_empty()) {} }
+        unsafe { critical_section::with(|_| RX_QUEUE.pop_front().unwrap()) }
+    }
+
+    fn send(&mut self, bytes: &[u8]) {
+        self.write_bytes(bytes).unwrap()
+    }
+}
+
+#[cfg(feature = "esp32c3")]
+#[interrupt]
+unsafe fn USB_SERIAL_JTAG() {
```

Comment:
I'm fine doing it in another PR, but there interrupt names should be aligned in the PACs then updated here.

Reply:
I'll do it before merging this, or else I'll forget about it - will also benefit esp-hal.